Ooui/Element.cs

```diff
@@ -9,6 +9,8 @@ public abstract class Element : Node
     {
         readonly Dictionary<string, object> attributes = new Dictionary<string, object> ();
 
+        readonly Dictionary<string, object> localAttributes = new Dictionary<string, object> ();
```

Comment:
Another approach is to merge the two dictionaries and have metadata determining if it's local or not.

To save memory the local attributes dictionary could be created on demand.